bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/processors/checksum/ChecksumVerifier.java

```diff
@@ -34,13 +34,16 @@ final public class ChecksumVerifier extends MessageDigestProcessingStep {
 	final private String providerName;
 	final private String algorithmId;
 	private final boolean insecureAlgorithm;
+	private int priority;
 
 	// public to access from tests
-	public ChecksumVerifier(String digestAlgorithm, String providerName, String algorithmId, boolean insecure) {
+	public ChecksumVerifier(String digestAlgorithm, String providerName, String algorithmId, boolean insecure,
+			int priority) {
```

Comment:
If there was any expectation that someone might be using this class directly, leaving the original constructor to call this new one with defaults would be good.  But I doubt there are such...

Reply:
Done.

Reply:
Thanks!

Reply:
Would it be useful to mark that constructor for removal as well?
The new one that is public for tests only could also be marked as deprecated (but not for removal), that that anybody that calls it, without reading the comment, gets it strikedout in the IDE.

Reply:
It is internal.. so actually we can delete, modify, .. any time.